### PR TITLE
feat: add collision_checker

### PR DESCRIPTION
**Pull request by 0x126**
## Description
- Added a new module for detecting collisions with objects
  - When an object is detected within a specific distance from the ego, it is considered a collision and outputs an ERROR diag

## Related links
- https://github.com/tier4/autoware_launch.x2/pull/263
- [TIER IV INTERNAL LINK](https://tier4.atlassian.net/browse/T4PB-26722)

## Tests performed
- Merge https://github.com/tier4/autoware_launch.x2/pull/263
- Manually placed objects like pedestrians in the planning_simulator, and confirmed that the output varies depending on the distance
  - Confirm ERROR is not output when operation mode is not AUTONOMOUS 
Screenshot from 2023-03-21 16-58-04

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/collision_checker/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+| `/perception/obstacle_segmentation/pointcloud` | `sensor_msgs::msg::PointCloud2`                        | Pointcloud of obstacles which the ego-vehicle should stop or avoid |
+| `/perception/object_recognition/objects`       | `autoware_auto_perception_msgs::msg::PredictedObjects` | Dynamic objects                                                    |
+| `/tf`                                          | `tf2_msgs::msg::TFMessage`                             | TF                                                                 |
+| `/tf_static`                                   | `tf2_msgs::msg::TFMessage`                             | TF static                                                          |
+
+### Output
+
+| Name           | Type                                    | Description |
+| -------------- | --------------------------------------- | ----------- |
+| `/diagnostics` | `diagnostic_msgs::msg::DiagnosticArray` | Diagnotics  |
```
Comment (TomohitoAndo) [inline planning/collision_checker/README.md:37]: typo `Diagnotics` -> `Diagnostics`

File: planning/collision_checker/include/collision_checker/node.hpp [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+private:
+  void checkCollision(diagnostic_updater::DiagnosticStatusWrapper & stat);
+
+  void onPointCloud(const sensor_msgs::msg::PointCloud2::ConstSharedPtr msg);
+
+  void onDynamicObjects(const PredictedObjects::ConstSharedPtr msg);
+
+  void onOperationMode(const OperationModeState::ConstSharedPtr msg);
+
+  // void onOdometry(const nav_msgs::msg::Odometry::ConstSharedPtr msg);
```
Comment (TomohitoAndo) [inline planning/collision_checker/include/collision_checker/node.hpp:67]: Please remove the unnecessary comments.

**Comment by TomohitoAndo** [comment]
I confirmed this works.

https://user-images.githubusercontent.com/11865769/231734059-ddd72ee2-61ac-4fbe-a334-1fccf929032d.mp4

File: planning/collision_checker/README.md (hunk 1 above)
Comment (0x126) [reply planning/collision_checker/README.md]: https://github.com/tier4/autoware.universe/pull/332/commits/0e209000fa6d269cde4348043da59ba3b547c842

File: planning/collision_checker/include/collision_checker/node.hpp (hunk 2 above)
Comment (0x126) [reply planning/collision_checker/include/collision_checker/node.hpp]: https://github.com/tier4/autoware.universe/pull/332/commits/a2d42bf5eee6d410b683716284cfd3c962f7f255

**Review by TomohitoAndo — APPROVED** [review]
LGTM

**Pull request merged**